graphics/src/compositor.rs

```diff
@@ -59,6 +59,18 @@ pub trait Compositor: Sized {
         background_color: Color,
         overlay: &[T],
     ) -> Result<(), SurfaceError>;
+
+    /// Renders the current [`Renderer`] primitives to an offscreen texture, and returns the bytes of
+    /// the texture.
+    ///
+    /// [`Renderer`]: Self::Renderer;
+    fn render_offscreen<T: AsRef<str>>(
+        &mut self,
+        renderer: &mut Self::Renderer,
+        viewport: &Viewport,
+        background_color: Color,
+        overlay: &[T],
```

Comment:
Do we care about rendering the debug overlay via this?

Reply:
Yeah I think it makes sense. It's as simple as toggling it off if you don't want it included